File: core/src/ru/myitschool/dcrawler/entities/skills/patterns/CharacterTargetPattern.java
package ru.myitschool.dcrawler.entities.skills.patterns;

import ru.myitschool.dcrawler.dungeon.DungeonCell;
import ru.myitschool.dcrawler.dungeon.DungeonMap;
import ru.myitschool.dcrawler.entities.skills.Skill;
import ru.myitschool.dcrawler.entities.skills.Target;

public class CharacterTargetPattern extends TargetPattern {

    public CharacterTargetPattern(Skill skill) {
        super(skill);
    }

    @Override
    public Target createTarget(Target target) {
        DungeonCell cell = DungeonMap.getCell(target.getX(), target.getY());
        if (cell.hasEntity() && cell.getEntity().isCharacter()){
            return target;
        }
        return null;
    }
}

File: core/src/com/wolg_vlad/dcrawler/entities/skills/patterns/CharacterTargetPattern.kt
package com.wolg_vlad.dcrawler.entities.skills.patterns

import com.wolg_vlad.dcrawler.dungeon.DungeonMap
import com.wolg_vlad.dcrawler.entities.skills.Skill
import com.wolg_vlad.dcrawler.entities.skills.Target

class CharacterTargetPattern(skill: Skill) : TargetPattern(skill) {
    override fun createTarget(target: Target): Target {
        return target
    }

    override fun isValid(target: Target): Boolean {
        val cell = DungeonMap.getCell(target.checkX, target.checkY)
        return cell?.entity?.isCharacter?:false
    }
}